client/src/Main.elm

```diff
@@ -1277,6 +1365,63 @@ getAuthentication auth =
         }
 
 
+getVectors : Auth -> Cmd Msg
+getVectors auth =
+    case auth of
+        Authenticated usr ->
+            authenticatedGet usr.token "http://localhost:8000/vectors/" VectorsReceived vectorDecoder
+
+        _ ->
+            Cmd.none
+
+
+createVector : String -> Vector -> Cmd Msg
+createVector token vector =
+    case vector of
+        Level0Vec vec ->
+            Http.request
+                { method = "POST"
+                , headers = [ Http.header "Authorization" ("Bearer " ++ token) ]
+                , url = "http://localhost:8000/vectors/"
+                , body = Http.jsonBody (level0Encoder vec)
+                , expect = Http.expectJson Level0Created level0Decoder
+                , timeout = Nothing
+                , tracker = Nothing
+                }
+
+        BackboneVec vec ->
+            Http.request
+                { method = "POST"
+                , headers = [ Http.header "Authorization" ("Bearer " ++ token) ]
+                , url = "http://localhost:8000/vectors/"
+                , body = Http.jsonBody (backboneEncoder vec)
+                , expect = Http.expectJson BackboneCreated backboneDecoder
+                , timeout = Nothing
+                , tracker = Nothing
+                }
+
+        LevelNVec vec ->
+            Http.request
+                { method = "POST"
+                , headers = [ Http.header "Authorization" ("Bearer " ++ token) ]
+                , url = "http://localhost:8000/vectors/"
+                , body = Http.jsonBody (levelNEncoder vec)
+                , expect = Http.expectJson Level1Created level1Decoder
+                , timeout = Nothing
+                , tracker = Nothing
+                }
+
+
+postVectors : Auth -> Vector -> Cmd Msg
+postVectors auth vector =
+    case auth of
+        Authenticated usr ->
+            createVector usr.token vector
+
+        _ ->
+            Cmd.none
```

Comment:
With the above, this can be removed. `createVector` is a better name than `postVector` since `postVector` "leaks" the underlying abstraction (that we're using the HTTP "POST" verb).